digdag-standards/src/main/java/io/digdag/standards/operator/td/TDOperator.java

```diff
@@ -408,17 +415,27 @@ private static int exponentialBackoffInterval(DurationInterval pollInterval, int
     static boolean isDeterministicClientException(Exception ex)
     {
         if (ex instanceof TDClientHttpException) {
-            int statusCode = ((TDClientHttpException) ex).getStatusCode();
-            switch (statusCode) {
-                case HttpStatus.TOO_MANY_REQUESTS_429:
-                case HttpStatus.REQUEST_TIMEOUT_408:
-                    return false;
-                default:
-                    // return true if 4xx
-                    return statusCode >= 400 && statusCode < 500;
-            }
+            return isAuthenticationErrorException((TDClientHttpException)ex);
         }
-        else if (ex instanceof TDClientException) {
+        return isFailedBeforeSendClientException(ex);
+    }
+
+    static boolean isAuthenticationErrorException(TDClientHttpException ex)
+    {
+        int statusCode = ex.getStatusCode();
+        switch (statusCode) {
+            case HttpStatus.TOO_MANY_REQUESTS_429:
+            case HttpStatus.REQUEST_TIMEOUT_408:
+                return false;
+            default:
+                // return true if 4xx
+                return statusCode >= 400 && statusCode < 500;
```

Comment:
Will it be better to use httpstatus code instead of 400 ~ 500 ? because not all 4xx are authentication failure

Reply:
Thanks for pointing out! I fixed here. https://github.com/treasure-data/digdag/pull/1362/commits/73cdc06b428726998469a599acc253af00472d2e